trin-core/src/utils/content_key.rs

```diff
@@ -21,12 +22,16 @@ impl VecToArrayError {
     }
 }
 
-pub fn keccak(key: &str) -> Result<U256, VecToArrayError> {
+pub fn keccak256(key: &[u8]) -> U256 {
     let mut hasher = Keccak256::new();
     hasher.update(key);
-    let result = hasher.finalize().to_vec();
-    // if the underlying hash library is secure, then a 32 len vector will always be returned, so no error should ever occur
-    Ok(U256::from(vec_to_array::<u8, 32>(result)?))
+    let result = &hasher.finalize()[..];
+    U256::from(result)
+}
+
+pub fn sha256(key: &[u8]) -> U256 {
```

Comment:
Would it be possible to switch this out for something more similar to https://github.com/ethereum/trin/blob/master/trin-core/src/bin/generate-data.rs#L77 ? The fewer dependencies the better, so if we can avoid a redundant import in this case (and remove the `hmac_sha256` dependency entirely) that seems preferable

Reply:
I would actually prefer it if the `sha3` dependency was removed and we keep the one I have. The hash function that you linked uses sha3_256, from the `sha3` family whereas the one I'm using is from the `sha2` family. I remember talking about this in some detail, that we were specifically going to use the one from the `sha2` family. The one from `sha3` was modified by NIST.